pkgs/development/guile-modules/guile-lib/default.nix

```diff
@@ -16,13 +17,20 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [
+    autoreconfHook
     pkg-config
   ];
   buildInputs = [
     guile
     texinfo
   ];
 
+  postPatch = ''
+    substituteInPlace configure.ac \
+      --replace 'SITEDIR="$datadir/guile-lib"' 'SITEDIR=$datadir/guile/site/$GUILE_EFFECTIVE_VERSION' \
+      --replace 'SITECCACHEDIR="$libdir/guile-lib/guile/$GUILE_EFFECTIVE_VERSION/site-ccache"' 'SITECCACHEDIR="$libdir/guile/$GUILE_EFFECTIVE_VERSION/site-ccache"'
+  '';
```

Comment:
Any idea why only we(nixpkgs) need this change?

Reply:
There's a `--with-guile-site` configure option that gentoo and netbsd packages are using

Reply:
> Any idea why only we(nixpkgs) need this change?

guile-lib module is placed at `$out/share/guile-lib` which is not part of Guile's search path (`$out/share/guile/$GUILE_EFFECTIVE_VERSION/site`).

> There's a --with-guile-site configure option that the gentoo and netbsd packages are using. can you try it instead of substituting?

`--with-guile-site` makes it so the module is installed at Guile's store path instead of the package's own output path. Basically, not possible.

Reply:
Looks like pkg-config is used to get the build vars

Please try
```
PKG_CONFIG_GUILE_${builtins.replaceStrings ["."] ["_"] guile.effectiveVersion}_SITEDIR = "${placeholder "out"}/${guile.siteDir}";
PKG_CONFIG_GUILE_${builtins.replaceStrings ["."] ["_"] guile.effectiveVersion}_SITECCACHEDIR = "${placeholder "out"}/${guile.siteCcacheDir}";
```
(Typed on phone so there may be mistakes)

With the previously mentioned configure option

Reply:
It still attempts to install it through Guile's store path.

```
warning: Git tree '/home/foo-dogsquared/library/projects/packages/nixpkgs' is dirty
error: builder for '/nix/store/wchb4hn0rnagm4hkqr77lfjpbp12i8i6-guile-lib-0.2.7.drv' failed with exit code 2;
       last 10 log lines:
       > make[1]: Entering directory '/build/guile-lib-0.2.7/src'
       > make[2]: Entering directory '/build/guile-lib-0.2.7/src'
       > make[2]: Nothing to be done for 'install-exec-am'.
       >  /nix/store/apn3p2b40xvirn7w740wv2gy330ppib5-coreutils-9.3/bin/mkdir -p '/nix/store/71rkkgf3yxa990f3qg44lqzb2vgpqp9x-guile-3.0.9/share/guile/site'
       > /nix/store/apn3p2b40xvirn7w740wv2gy330ppib5-coreutils-9.3/bin/mkdir: cannot create directory '/nix/store/71rkkgf3yxa990f3qg44lqzb2vgpqp9x-guile-3.0.9/share/guile/site': Permission denied
       > make[2]: *** [Makefile:491: install-nobase_modDATA] Error 1
       > make[2]: Leaving directory '/build/guile-lib-0.2.7/src'
       > make[1]: *** [Makefile:611: install-am] Error 2
       > make[1]: Leaving directory '/build/guile-lib-0.2.7/src'
       > make: *** [Makefile:480: install-recursive] Error 1
       For full logs, run 'nix log /nix/store/wchb4hn0rnagm4hkqr77lfjpbp12i8i6-guile-lib-0.2.7.drv'.
```

Just to be sure, I also tried to build it without the `PKG_CONFIG` prefix and still the same result. For complete reference, here's the diff of the suggested changes at the time of building it.

```diff
diff --git a/pkgs/development/guile-modules/guile-lib/default.nix b/pkgs/development/guile-modules/guile-lib/default.nix
index 75e45b00d4e7..9eda51a74716 100644
--- a/pkgs/development/guile-modules/guile-lib/default.nix
+++ b/pkgs/development/guile-modules/guile-lib/default.nix
@@ -25,11 +25,9 @@ stdenv.mkDerivation rec {
     texinfo
   ];

-  postPatch = ''
-    substituteInPlace configure.ac \
-      --replace 'SITEDIR="$datadir/guile-lib"' 'SITEDIR=$datadir/guile/site/$GUILE_EFFECTIVE_VERSION' \
-      --replace 'SITECCACHEDIR="$libdir/guile-lib/guile/$GUILE_EFFECTIVE_VERSION/site-ccache"' 'SITECCACHEDIR="$libdir/guile/$GUILE_EFFECTIVE_VERSION/site-ccache"'
-  '';
+  "PKG_CONFIG_GUILE_${builtins.replaceStrings ["."] ["_"] guile.effectiveVersion}_SITEDIR" = "${placeholder "out"}/${guile.siteDir}";
+  "PKG_CONFIG_GUILE_${builtins.replaceStrings ["."] ["_"] guile.effectiveVersion}_SITECCACHEDIR" = "${placeholder "out"}/${guile.siteCcacheDir}";
+  configureFlags = [ "--with-guile-site=yes" ];

   makeFlags = [ "GUILE_AUTO_COMPILE=0" ];
```

Reply:
```
$ PKG_CONFIG_PATH="$(nix build 'nixpkgs#guile^dev' --print-out-paths)/lib/pkgconfig" PKG_CONFIG_GUILE_3_0_SITEDIR=/sometest , pkg-config guile-3.0 --variable=sitedir
/sometest
```

The override works as it should but looks like instead of using the pkg-config stuff in the guile.m4, it's using the functions in guile-ext.m4 which use $GUILE to get the variables other than `GUILE_SITE_DIR` :(